nerfstudio/data/utils/nerfstudio_collate.py

```diff
@@ -163,11 +163,38 @@ def nerfstudio_collate(batch: Any, extra_mappings: Union[Dict[type, Callable], N
             op = torch.cat
 
         # Create metadata dictionary
-        metadata_keys = batch[0].metadata.keys()
-        assert all(
-            (cam.metadata.keys() == metadata_keys for cam in batch)
-        ), "All cameras must have the same metadata keys."
-        metadata = {key: op([cam.metadata[key] for cam in batch], dim=0) for key in metadata_keys}
+        if batch[0].metadata is not None:
+            metadata_keys = batch[0].metadata.keys()
+            assert all(
+                (cam.metadata.keys() == metadata_keys for cam in batch)
+            ), "All cameras must have the same metadata keys."
+            metadata = {key: op([cam.metadata[key] for cam in batch], dim=0) for key in metadata_keys}
+        else:
+            assert all((cam.metadata is None for cam in batch)), "All cameras must have the same metadata keys."
+            metadata = None
+
+        if batch[0].distortion_params is not None:
+            assert all(
+                (cam.distortion_params is not None for cam in batch)
+            ), "All cameras must have distortion present or absent."
+
+            distortion_params = op(
+                [cameras.distortion_params for cameras in batch],
+                dim=0,
+            )
+        else:
+            assert all(
+                (cam.distortion_params is None for cam in batch)
+            ), "All cameras must have distortion present or absent."
+            distortion_params = None
+
+        if batch[0].times is not None:
+            assert all((cam.times is not None for cam in batch)), "All cameras must have times present or absent."
```

Comment:
Can we move this assert to the beginning of the function just like what we did for distortion_params?